### ISSUE TITLE
[pathlib] Option so that OSError does not block glob in pathlib library

### ISSUE THREAD

**Issue by 519e1874-27a7-4f51-83ff-223a8006c36f** [python/cpython#90208]
BPO | [46050](https://bugs.python.org/issue46050)
--- | :---
Nosy | @matt32106

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2021-12-11.16:25:35.541>
labels = ['3.8', 'type-feature', 'library']
title = '[pathlib] Option so that OSError does not block glob in pathlib library'
updated_at = <Date 2021-12-11.16:25:35.541>
user = 'https://github.com/matt32106'
```

bugs.python.org fields:
```python
activity = <Date 2021-12-11.16:25:35.541>
actor = 'matt32106'
assignee = 'none'
closed = False
closed_date = None
closer = None
components = ['Library (Lib)']
creation = <Date 2021-12-11.16:25:35.541>
creator = 'matt32106'
dependencies = []
files = []
hgrepos = []
issue_num = 46050
keywords = []
message_count = 1.0
messages = ['408314']
nosy_count = 1.0
nosy_names = ['matt32106']
pr_nums = []
priority = 'normal'
resolution = None
stage = None
status = 'open'
superseder = None
type = 'enhancement'
url = 'https://bugs.python.org/issue46050'
versions = ['Python 3.8']
```

</p></details>

### Linked PRs
* gh-104141

**Comment by barneygale** [comment]
IMO it should ignore all `OSError` exceptions that are generated below the top-level path. PR incoming...

**Comment by barneygale** [comment]
Fixed in #104141 / [94f30c7](https://github.com/python/cpython/commit/94f30c75576bb8a20724b2ac758fa33af089a522)
